cmd/controller/controller.go

```diff
@@ -190,38 +301,45 @@ func (c *Controller) updateRelease(key string) error {
 
 	helmObj := obj.(*helmCrdV1.HelmRelease)
 
-	// FIXME: make configurable
-	keyring := "/keyring/pubring.gpg"
-
-	dl := downloader.ChartDownloader{
-		HelmHome: settings.Home,
-		Out:      os.Stdout,
-		Keyring:  keyring,
-		Getters:  getter.All(settings),
-		Verify:   downloader.VerifyNever, // FIXME
-	}
-
 	repoURL := helmObj.Spec.RepoURL
 	if repoURL == "" {
 		// FIXME: Make configurable
 		repoURL = defaultRepoURL
 	}
+	repoURL = path.Join(strings.TrimSuffix(strings.TrimSpace(repoURL), "/index.yaml"), "index.yaml")
```

Comment:
`path.Join` will mangle URLs, since it tries to "clean up" what it thinks is a simple /-separated path.  See https://play.golang.org/p/1zlsxjvQJDQ for example.

You need to either do the string manipulation yourself, or parse the URL and use `path.Join` only the *path* component of the URL.  In this case, I suggest just doing `repoURL = strings.TrimSuffix(repoURL, "/") + "/index.yaml"`